# CVE-2012-2135


## Metadata
| Published | Last Modified | Assigner | Version | Format |
| --------- | ------------- | -------- | ------- | ------ |
| 2012-08-14 22:55:00 | 2022-07-28 12:36:00 | secalert@redhat.com | 4.0 | MITRE | 

## Description
| Language | Description |
| -------- | ----------- |
| en | The utf-16 decoder in Python 3.1 through 3.3 does not update the aligned_end variable after calling the unicode_decode_call_errorhandler function, which allows remote attackers to obtain sensitive information (process memory) or cause a denial of service (memory corruption and crash) via unspecified vectors. |

## Configurations
| Vulnerable | CPE 23 URI | CPE Name | Operator |
| ---------- | ---------- | -------- | -------- |
| True | cpe:2.3:a:python:python:*:*:*:*:*:*:*:* | [] | OR |
| True | cpe:2.3:a:python:python:*:*:*:*:*:*:*:* | [] | OR |
| True | cpe:2.3:a:python:python:*:*:*:*:*:*:*:* | [] | OR |
| True | cpe:2.3:o:canonical:ubuntu_linux:11.04:*:*:*:*:*:*:* | [] | OR |
| True | cpe:2.3:o:canonical:ubuntu_linux:11.10:*:*:*:*:*:*:* | [] | OR |
| True | cpe:2.3:o:canonical:ubuntu_linux:12.10:*:*:*:*:*:*:* | [] | OR |
| True | cpe:2.3:o:canonical:ubuntu_linux:10.04:*:*:*:-:*:*:* | [] | OR |
| True | cpe:2.3:o:canonical:ubuntu_linux:12.04:*:*:*:-:*:*:* | [] | OR |
| True | cpe:2.3:o:debian:debian_linux:6.0:*:*:*:*:*:*:* | [] | OR |

## CVSSv3 Information
*** Not Provided/Calculated ***

## CVSSv2 Information
| Score | Severity | Impact Score | Exploitability Score | Vector String | Version | Access Vector | Access Complexity | Authentication | User Interaction Required | Obtain All Privilge | Obtain User Privilege | Obtain Other Privilege | Confidentiality Impact | Integrity Impact | Availability Impact |
| ----- | -------- | ------------ | -------------------- | ------------- | ------- | ------------- | ----------------- | -------------- | ------------------------- | ------------------- | --------------------- | ---------------------- | ---------------------- | ---------------- | ------------------- |
| 6.4 | MEDIUM | 4.9 | 10.0 | AV:N/AC:L/Au:N/C:P/I:N/A:P | 2.0 | NETWORK | LOW | NONE | 0 | False | False | False | PARTIAL | NONE | PARTIAL |

## Problem Type
NVD-CWE-Other

## References
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
http://bugs.python.org/issue14579|http://bugs.python.org/issue14579|MISC|['Vendor Advisory']
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
[oss-security] 20120425 Re: CVE Request: Python 3.2/3.3 utf-16 decoder unicode_decode_call_errorhandler aligned_end is not updated|http://www.openwall.com/lists/oss-security/2012/04/25/4|MLIST|['Mailing List', 'Third Party Advisory']
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
[oss-security] 20120425 CVE Request: Python 3.2/3.3 utf-16 decoder unicode_decode_call_errorhandler aligned_end is not updated|http://www.openwall.com/lists/oss-security/2012/04/25/2|MLIST|['Mailing List', 'Third Party Advisory']
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=670389|http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=670389|MISC|['Third Party Advisory']
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
USN-1615-1|http://www.ubuntu.com/usn/USN-1615-1|UBUNTU|['Third Party Advisory']
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
51089|http://secunia.com/advisories/51089|SECUNIA|['Third Party Advisory']
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
USN-1616-1|http://www.ubuntu.com/usn/USN-1616-1|UBUNTU|['Third Party Advisory']
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
51087|http://secunia.com/advisories/51087|SECUNIA|['Third Party Advisory']
